lib/index.js

```diff
@@ -32,37 +42,76 @@ function validate(validators, filteredValue) {
   });
 }
 
-function createValidations(schema, values) {
+/**
+ * Parse the values object and map validation rules to each field.
+ * @param {string} path Nested parent key of object. e.g. "outerKey.innerKey".
+ * @param {object} schema Object describing the schema to validate against.
+ * @param {object} values Object which is validated against the schema.
+ * @returns {void}
+ */
+function createValidations(path, schema, values) {
   const schemaKeys = Object.keys(schema);
   schemaKeys.map(key => {
     const schemaItem = schema[key];
     const value = values[key];
 
-    const validateThing = valueToValidate => {
-      return validate(schemaItem.validators, valueToValidate).then(result => {
-        result.key = key;
+    let nestedKey = key;
+    if (path !== null) {
+      nestedKey = `${path}.${key}`;
+    }
+
+    const validateThing = valueToValidate =>
+      validate(schemaItem.validators, valueToValidate).then(result => {
+        result.key = nestedKey;
         return result;
       });
-    };
 
     if (isEmpty(value)) {
       if ((typeof schemaItem.empty === 'undefined') || (typeof schemaItem.empty.default === 'undefined')) {
         throw Error(`No default set for ${key}`);
       } else {
-        promises.push(Promise.resolve({key, valid: true, validValue: schemaItem.empty.default}));
+        promises.push(Promise.resolve({key: nestedKey, valid: true, validValue: schemaItem.empty.default}));
       }
     } else if (typeof schemaItem.filters !== 'undefined' && schemaItem.filters.length > 0) {
       const chain = filterChain(schemaItem.filters);
       promises.push(applyFilterToField(chain, value).then(filteredValue => (validateThing(filteredValue))));
+    } else if (schemaItem.children) {
+      createValidations(nestedKey, schemaItem.children, value);
     } else {
-      if (schemaItem.children) {
-        createValidations(schemaItem.children, value);
-      }
       promises.push(validateThing(value));
     }
   });
 }
 
+/**
+ * Adds a value with a nested key (dot notation) to an object, creating
```

Comment:
Does this need tests? Should we use something like [object-path](https://www.npmjs.com/package/object-path) which is already has a comprehensive test suite?